src/main/java/com/dev/cinema/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,68 @@
+package com.dev.cinema.controller;
+
+import com.dev.cinema.model.MovieSession;
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.Ticket;
+import com.dev.cinema.model.User;
+import com.dev.cinema.model.dto.ShoppingCartResponseDto;
+import com.dev.cinema.model.dto.TicketResponseDto;
+import com.dev.cinema.service.MovieSessionService;
+import com.dev.cinema.service.ShoppingCartService;
+import com.dev.cinema.service.UserService;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final MovieSessionService movieSessionService;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+
+    public ShoppingCartController(MovieSessionService movieSessionService,
+                                  ShoppingCartService shoppingCartService,
+                                  UserService userService) {
+        this.movieSessionService = movieSessionService;
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+    }
+
+    @GetMapping("/by-user")
+    public ShoppingCartResponseDto getCartByUser(@RequestParam Long userId) {
+        User user = userService.findById(userId);
+        ShoppingCart shoppingCart = shoppingCartService.getByUser(user);
+        return convertToShoppingCartDto(shoppingCart);
+    }
+
+    @PostMapping("/add-movie-session")
```

Comment:
and here(p.s I know that didn’t  notice this during previous review)))
```suggestion
    @PostMapping
```